tree/dataframe/test/dataframe_datasetspec.cxx

```diff
@@ -0,0 +1,449 @@
+   // this garbage is of no interest to be tested
+   for (auto i = 0u; i < ranges.size(); ++i) {
+      for (const auto &s : specs[i]) {
+         // nested structure so that the event loop is not called multiple times with different actions
+         std::vector<ROOT::RDF::RResultPtr<std::vector<ULong64_t>>> friends;
+         friends.push_back(RDataFrame(RDatasetSpec(s)).Take<ULong64_t>("friendTree.z"));
+         // friends.push_back(RDataFrame(RDatasetSpec(s)).Take<ULong64_t>("friendShortTree.y"));
+         friends.push_back(RDataFrame(RDatasetSpec(s)).Take<ULong64_t>("friendChain1.z"));
+         // friends.push_back(RDataFrame(RDatasetSpec(s)).Take<ULong64_t>("friendShortChain1.z"));
+         friends.push_back(RDataFrame(RDatasetSpec(s)).Take<ULong64_t>("friendChainN.z"));
+         // friends.push_back(RDataFrame(RDatasetSpec(s)).Take<ULong64_t>("friendShortChainN.z"));
+
+         for (auto &fr : friends) {
+            std::sort((*fr).begin(), (*fr).end());
+            EXPECT_VEC_EQ(*fr, expectedRess[i]);
+         }
+      }
+   }
+}
+
+TEST_P(RDatasetSpecTest, Histo1D)
+{
+   RDatasetSpec spec(
+      {{"subTree0"s, "specTestFile8.root"s}, {"subTree1"s, "specTestFile9.root"s}}); // vertical concatenation
+   spec.AddFriend(
+      {{"subTree2"s, "specTestFile10.root"s}, {"subTree3"s, "specTestFile11.root"s}}); // horizontal concatenation
+   ROOT::RDataFrame d(spec);
+
+   auto h1 = d.Histo1D(::TH1D("h1", "h1", 10, 0, 10), "x");
+   auto h2 = d.Histo1D({"h2", "h2", 10, 0, 10}, "x");
+   auto h1w = d.Histo1D(::TH1D("h0w", "h0w", 10, 0, 10), "x", "w");
+   auto h2w = d.Histo1D({"h2w", "h2w", 10, 0, 10}, "x", "w");
+   std::vector<double> edgesd{1, 2, 3, 4, 5, 6, 10};
+   auto h1edgesd = d.Histo1D(::TH1D("h1edgesd", "h1edgesd", (int)edgesd.size() - 1, edgesd.data()), "x");
+   auto h2edgesd = d.Histo1D({"h2edgesd", "h2edgesd", (int)edgesd.size() - 1, edgesd.data()}, "x");
+
+   ROOT::RDF::TH1DModel m0("m0", "m0", 10, 0, 10);
+   ROOT::RDF::TH1DModel m1(::TH1D("m1", "m1", 10, 0, 10));
+
+   auto hm0 = d.Histo1D(m0, "x");
+   auto hm1 = d.Histo1D(m1, "x");
+   auto hm0w = d.Histo1D(m0, "x", "w");
+   auto hm1w = d.Histo1D(m1, "x", "w");
+
+   std::vector<double> ref({0., 1., 2., 3., 4., 5., 6., 7., 8., 9., 10.});
+
+   auto CheckBins = [](const TAxis *axis, const std::vector<double> &v) {
+      auto nBins = axis->GetNbins();
+      auto nBinsp1 = nBins + 1;
+      auto nBinsm1 = nBins - 1;
+      EXPECT_EQ(nBinsp1, (int)v.size());
+      for (auto i : ROOT::TSeqI(1, nBinsp1)) {
+         EXPECT_DOUBLE_EQ(axis->GetBinLowEdge(i), (double)v[i - 1]);
+      }
+      EXPECT_DOUBLE_EQ(axis->GetBinUpEdge(nBinsm1), (double)v[nBinsm1]);
+   };
+
+   CheckBins(h1->GetXaxis(), ref);
+   CheckBins(h2->GetXaxis(), ref);
+   CheckBins(hm0->GetXaxis(), ref);
+   CheckBins(hm1->GetXaxis(), ref);
+   CheckBins(h1edgesd->GetXaxis(), edgesd);
+   CheckBins(h2edgesd->GetXaxis(), edgesd);
+}
+
+TEST_P(RDatasetSpecTest, FilterDependingOnVariation)
```

Comment:
testing same as below:
I aimed at checking interaction of the RDF created with spec with some common actions. Maybe the spec was setting something "unexpected" for the RLoopManager. Now notice that one of the columns came from a friend.

Reply:
in practice this test case does not add anything on top of the one that checks that we read the correct values with `Take`. but I know that because I know exactly how the internals work, and how exactly the internals work might change, so ok let's keep this (and the other one).

Reply:
in practice this test case does not add anything on top of the one that checks that we read the correct values with `Take`. but I know that because I know exactly how the internals work, and how exactly the internals work might change, so ok let's keep this (and the other one).